rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/model/ParametricSummaryBridge.java

```diff
@@ -167,6 +167,18 @@ private void replaceFuture(InterruptibleFuture<Lazy<IRangeMap<T>>> newFuture) {
         abstract InterruptibleFuture<T> requestDedicated(Position r);
 
         public InterruptibleFuture<T> lookup(Position cursor) {
+            // If an analyzer is available, use it to compute the summary for
+            // lookups (if needed). Else, if a builder is available, use it.
+            // Else, fail. The following code builds this logic inside-out.
+            InterruptibleFuture<T> lookupFail = InterruptibleFuture.completedFuture(empty);
```

Comment:
ouch, this is painfull stack, did you experiment with other ways to chain these? doesn't this always run the builder? also if the analyze is already succeeded? as in, we might be causing a lot more invocations (and possible races between them?)

If we stop running the builder outside of the save context, it might make this code less painful?

Reply:
Greatly simplified requirements, so greatly simplified code